teammapper-backend/src/map/controllers/maps.controller.ts

```diff
@@ -49,4 +49,27 @@ export default class MapsController {
       modificationSecret: newMap.modificationSecret,
     }
   }
+
+  @Post(':id/share')
+  async duplicate(
+    @Param('id') mapId: string,
+  ): Promise<IMmpClientPrivateMap> {
+    const oldMap = await this.mapsService.findMap(mapId).catch((e: Error) => {
+      if (e.name === 'MalformedUUIDError') throw new NotFoundException()
+    })
+
+    if (!oldMap) return Promise.reject()
```

Comment:
not found exception might be better suited here

Reply:
👍